### PR TITLE
Csharp updates for 4, 9 chapters

### PR DESCRIPTION
**Pull request by ManSnowThe**
Fixed GetGCDList method in 06_GetGCD (Chapter 4).
Updated LongestCommonSubsequence method (Chapter 9).
Added Levenshtein Distance (Chapter 9).

### CODE REVIEW

File: 04_quicksort/csharp/06_GetGCD/program.cs [hunk 1]
```diff
@@ -4,28 +4,37 @@
...
 
         //Get great Comman Divisor
-        public static int GetGCD(int FirstNumber, int SecondNumber)
-            => SecondNumber == default ? FirstNumber : GetGCD(SecondNumber, FirstNumber % SecondNumber);
+        public static int GetGCD(int firstNumber, int secondNumber)
+            => secondNumber == default ? firstNumber : GetGCD(secondNumber, firstNumber % secondNumber);
 
         //Get great Comman Divisor of list
-        public static int GetGCDList(List<int> lst)
+        public static int GetGCDList(IEnumerable<int> lst)
```
Comment (egonSchiele) [inline 04_quicksort/csharp/06_GetGCD/program.cs:24]: What's the advantage of `IEnumerable` over `List`?

File: 09_dynamic_programming/csharp/02_levenshtein/Program.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+        public static int LevenshteinDistance(string source, string target)
+        {
+            var matrix = CreateMatrix(source, target);
+
+            for (int i = 1; i <= source.Length; i++)
+            {
+                for (int j = 1; j <= target.Length; j++)
+                {
+                    matrix[i, j] = Math.Min(matrix[i, j - 1] + 1, Math.Min(
```
Comment (egonSchiele) [inline 09_dynamic_programming/csharp/02_levenshtein/Program.cs:21]: this line is a bit hard to read, could you turn it into 2 - 3 lines that are easier to read?

File: 09_dynamic_programming/csharp/02_levenshtein/Program.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            return matrix[source.Length, target.Length];
+        }
+
+        private static int[,] CreateMatrix(string source, string target)
+        {
+            var matrix = new int[source.Length + 1, target.Length + 1];
+
+            if (source.Length < target.Length)
+            {
+                var tempSrc = source;
```
Comment (egonSchiele) [inline 09_dynamic_programming/csharp/02_levenshtein/Program.cs:35]: You don't need a temporary var I believe https://stackoverflow.com/a/39190792/20488510

File: 09_dynamic_programming/csharp/02_levenshtein/Program.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
+using System;
+
+namespace ConsoleApplication
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            var result = LevenshteinDistance("gibraltar", "labrador");
```
Comment (egonSchiele) [inline 09_dynamic_programming/csharp/02_levenshtein/Program.cs:9]: Actually, I like this file but don't think I would include it. I think the matrix it is building is cool, but not needed for this chapter.

**Review by egonSchiele — CHANGES_REQUESTED** [review]
couple changes, but looks good overall

File: 04_quicksort/csharp/06_GetGCD/program.cs (hunk 1 above)
Comment (ManSnowThe) [reply 04_quicksort/csharp/06_GetGCD/program.cs:24]: All collections are implementing `IEnumerable` interface and it's used to itterate through it. I used it that you can call method with any collection or itterable object, even simple array. Preferable use `IEnumerable` over `List` when you don't need modify collection. `IEnumerable` is read-only and `List` is not.

**Pull request merged**